public/app/themes/justice/inc/search.php

```diff
@@ -0,0 +1,213 @@
+        });
+    }
+
+    /**
+     * Check if the search query is empty.
+     *
+     * @return bool True if the search query is empty, false otherwise.
+     */
+
+    public function hasEmptyQuery(): bool
+    {
+        return empty(get_search_query());
+    }
+
+    /**
+     * Get the number of search results.
+     *
+     * @return int|null The number of search results.
+     */
+
+    public function getResultCount(): ?int
+    {
+        if (empty(get_search_query())) {
+            return null;
+        }
+
+        global $wp_query;
+        return $wp_query->found_posts;
+    }
+
+    /**
+     * Get the sort options for the search results.
+     *
+     * @return array An array of sort options.
+     */
+
+    public function getSortOptions(): array
+    {
+        $orderby = get_query_var('orderby');
+        return [
+            'relevance' => [
+                'label' => 'Relevance',
+                'url' => '/search/' . get_query_var('s'),
+                'selected' => empty($orderby) || $orderby === 'relevance',
+            ],
+            'date' => [
+                'label' => 'Most recent',
+                'url' => '/search/' . get_query_var('s') . '?orderby=date',
+                'selected' => $orderby === 'date',
+            ],
+        ];
+    }
+
+    public function getSuggestion(): ?string
+    {
+        if (empty(get_search_query())) {
+            return null;
+        }
+        return null;
+        // $q = get_search_query(false);
+        // error_log(print_r('$q: ' . $q, true));
+        // $s = \relevanssi_didyoumean($q, '', '', 5);
+        // error_log(print_r($s, true));
+        // return 'x';
+    }
+
+    /**
+     * Redirect old search URLs to the new search page.
+     *
+     * @return void
+     */
+
+    public function redirectOldSearchUrls()
+    {
+
+        $search_params = ['s' => null];
+
+        if (isset($_GET['s'])) {
+            // Redirect the s parameter to the new search page.
+            $search_params['s'] = $_GET['s'];
+        } else if (isset($_GET['query'])) {
+            // Redirect old search URLs to the new search page.
+            $search_params['s'] = $_GET['query'];
+        }
+
+        if (!$search_params['s']) {
+            return;
+        }
+
+        wp_redirect('/search/' . $search_params['s']);
+        exit;
+    }
+
+    /**
+     * Handle a search query with no search term.
+     *
+     * @param string $search The search query.
+     * @param \WP_Query $q The main WordPress query.
+     * @return string The modified search query.
+     */
+
+    public function handleEmptySearch($search, \WP_Query $q)
+    {
+        if (!is_admin() && empty($search) && $q->is_search() && $q->is_main_query()) {
+            // Return an empty search query to prevent any results from being returned.
+            $search .= " AND 0=1 ";
+        }
+
+        return $search;
+    }
+
+    // TODO: Redirect old query string search URLs to the new search page.
```

Comment:
Is this a work in progress?
Are you happy for this to remain for now?

Reply:
Good spot, yes, it's one of the next things on my list.